src/reporting/Market.sol

```diff
@@ -0,0 +1,604 @@
+// Copyright (C) 2015 Forecast Foundation OU, full GPL notice in LICENSE
+
+pragma solidity ^0.4.13;
+
+import 'ROOT/legacy_reputation/Ownable.sol';
+import 'ROOT/extensions/MarketExtensions.sol';
+import 'ROOT/reporting/Branch.sol';
+import 'ROOT/reporting/ReportingToken.sol';
+import 'ROOT/reporting/ReputationToken.sol';
+import 'ROOT/reporting/DisputeBondToken.sol';
+import 'ROOT/reporting/Interfaces.sol';
+import 'ROOT/trading/Cash.sol';
+import 'ROOT/libraries/DelegationTarget.sol';
+import 'ROOT/extensions/MarketFeeCalculator.sol';
+import 'ROOT/factories/MapFactory.sol';
+import 'ROOT/factories/ShareTokenFactory.sol';
+import 'ROOT/factories/ReportingTokenFactory.sol';
+import 'ROOT/factories/DisputeBondTokenFactory.sol';
+import 'ROOT/libraries/Typed.sol';
+import 'ROOT/libraries/Initializable.sol';
+import 'ROOT/libraries/token/ERC20Basic.sol';
+import 'ROOT/libraries/math/SafeMathUint256.sol';
+import 'ROOT/libraries/math/SafeMathInt256.sol';
+
+
+contract Market is DelegationTarget, Typed, Initializable, Ownable {
+    using SafeMathUint256 for uint256;
+    using SafeMathInt256 for int256;
+
+    enum ReportingState {
+        PRE_REPORTING,
+        AUTOMATED_REPORTING,
+        AUTOMATED_DISPUTE,
+        LIMITED_REPORTING,
+        LIMITED_DISPUTE,
+        ALL_REPORTING,
+        ALL_DISPUTE,
+        FORK,
```

Comment:
We need one more phase, `AWAITING_FINALIZATION` (name TBD).  It is the state where reporting is over and the market is ready to be finalized but it has not yet been finalized.  We _could_ make it so this function finalizes when in such a state, but that would mean `getReportingPhase` can't be `constant` anymore.  :/

I think for the sake of having a nice clean constant method we should go with having a phase that indicates finalizable, but not yet finalized.